pkg/providers/vsphere/reconciler/reconciler.go

```diff
@@ -213,6 +214,26 @@ func (r *Reconciler) ReconcileWorkers(ctx context.Context, log logr.Logger, spec
 	return clusters.ReconcileWorkersForEKSA(ctx, log, r.client, spec.Cluster, clusters.ToWorkers(w))
 }
 
+// InstallStorageClass install default storage class in workload cluster.
+func (r *Reconciler) InstallStorageClass(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	if clusterSpec.VSphereDatacenter.Spec.DisableCSI {
+		log.V(3).Info("VSphere CSI disabled, skipping storage class installation")
+		return controller.Result{}, nil
+	}
+
+	log = log.WithValues("phase", "reconcileStorageClass")
```

Comment:
I would put this at the top of the method
it guarantees all logs in this phase have these values